### PR TITLE
Input view model logic add

### PR DESCRIPTION
**Pull request by ERussel**
- input view model add: filter input characters, restrict by length, provide predicate to validate

### CODE REVIEW

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+public protocol InputViewModelPreprocessing {
```
Comment (unklayman) [inline SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: No docs.
I suggest to name it `TextPreprocessing`

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    func preprocess(text: String) -> String
+}
+
+public struct CompoundViewModelPreprocessor: InputViewModelPreprocessing {
```
Comment (unklayman) [inline SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: `CompoundTextPreprocessor `

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+public protocol InputViewModelPreprocessing {
+    func preprocess(text: String) -> String
```
Comment (unklayman) [inline SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: Why `preprocess` when it's processing text actually?)

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+public struct CompoundViewModelPreprocessor: InputViewModelPreprocessing {
+    public let children: [InputViewModelPreprocessing]
```
Comment (unklayman) [inline SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: [pre]processors

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift (hunk 1 above)
Comment (ERussel) [reply SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: Done.

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift (hunk 2 above)
Comment (ERussel) [reply SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: Done.

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift (hunk 3 above)
Comment (ERussel) [reply SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: Changed to ```process```

File: SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift (hunk 4 above)
Comment (ERussel) [reply SoraFoundation/Classes/ViewModel/Preprocessors/InputViewModelPreprocessor.swift]: Done.

**Review by unklayman — APPROVED** [review]

**Pull request merged**